test.js

```diff
@@ -447,3 +450,60 @@ test('alias - require previousId and userId', t => {
     })
   })
 })
+
+test.cb('e2e', t => {
+  const id = uid(16)
+
+  const testFn = function () {
+    var axiosClient = axios.create({
+      baseURL: 'https://api.runscope.com',
+      timeout: 10 * 1000,
+      headers: { 'Authorization': 'Bearer ' + process.env.RUNSCOPE_TOKEN }
```

Comment:
I’m worried about the implications this will have on open-source contributions, as folks won’t have access to this token.

Reply:
I think the best we can do here is to run this only on CI? If we don't automate this I'm sure we'll forget to run this test before releasing/making big changes.

Reply:
And I can't inline the token as it doesn't seem to be very well scoped. Though maybe we could setup a dedicated account for this and it wouldn't matter like we do for SauceLabs.

Reply:
Perhaps moving this test out of the "normal" test suite and _only_ having CI run it would solve the problem?

In `package.json`, we could do:

```json
{
  "scripts": {
    "test": "ava test.js",
    "test-e2e": "ava e2e/test.js
  }
}
```

And in `circle.yml`:

```yml
test:
  override:
    - yarn test
    - yarn test-e2e
```

Reply:
Instead of creating new folders and files, I propose we leave this test in `test.js` and use [is-ci](https://yarnpkg.com/en/package/is-ci) to determine whether to run it or not.

Reply:
@vadimdemedes That makes sense for the 90% use case, but what if you _want_ to run this locally?

Reply:
`$ CI=true npm test`

Reply:
Haha good point. Just seems a bit unintuitive to me. That will work for now tho =]

Reply:
Do we need a dependency for that? Seems like I could just guard it in `process.env.CI === true`?

Reply:
Better use `is-ci` to not care about that. All CIs have their specific environment variables, I don't think all of them have `CI` env var.

Reply:
Seems like an anti pattern to pull in a dependency to replace a small check like this one (a.k.a left-pad). `is-ci` itself is a one liner library, and depends on `ci-info` of which the relevant bits are 7 lines https://github.com/watson/ci-info/blob/master/index.js#L42-L48 (and really only 1-3  lines since we only care about Circle https://github.com/watson/ci-info/blob/master/index.js#L43).

And I think we don't really care about other CI systems? Both Travis/Circle (we're unlikely to use anything else in the near future) have CI=true https://circleci.com/docs/1.0/environment-variables/ https://docs.travis-ci.com/user/environment-variables/.

Reply:
@f2prateek you're thinking in go, not node. In node land, it's 100% acceptable to pull in a 1 line dependency.

Reply:
It's not about lines of code, but not doing something that's already been handled perfectly by other people. Left-pad example doesn't apply anymore, since npm took measures to prevent situations like that in the future. See https://github.com/sindresorhus/ama/issues/10#issuecomment-117766328 for a much more extensive answer to this.

Reply:
It still doesn't make sense to me to to pull in 50+ lines of dependencies when a one-liner does the job for us. I understand the flexibility is-ci/ci-info will offer, but seems overkill if we don't need that.

Reply:
But I'll put a pin in in this and use is-ci if that's what you'll want here.

Reply:
As a compromise, I propose we ditch CI check altogether. Instead, let's use something more descriptive of what's actually happening:

```js
if (process.env.INTEGRATION) {
  test(...)
}
```

```
$ INTEGRATION=true npm test
```

There should probably be a better name than `INTEGRATION` too.

Reply:
I very much prefer being explicit here and thought having to `CI=true yarn test` would have been awkward.